src/mail-app/mail/view/MailViewerHeader.ts

```diff
@@ -626,11 +626,12 @@ export class MailViewerHeader implements Component<MailViewerHeaderAttrs> {
 	}
 
 	private renderHardAuthenticationFailWarning(viewModel: MailViewerViewModel): Children | null {
-		const authFailed =
-			viewModel.checkMailAuthenticationStatus(MailAuthenticationStatus.HARD_FAIL) ||
+		const authFailedHard =
+			(!viewModel.checkMailAuthenticationStatus(MailAuthenticationStatus.AUTHENTICATED) &&
```

Comment:
Actually, I think my "improvement" will break if the mail is not properly loaded yet: https://github.com/tutao/tutanota/blob/53750be8cfd4c2dc8ba7b31b38cc578c4e3442f8/src/mail-app/mail/view/MailViewerViewModel.ts#L428
Maybe we should handle all enum values individually after all then. Either way, I think we should not ignore missing from headers.

Reply:
Fixed per https://github.com/tutao/tutanota/pull/8661/commits/f928ddc4c8ff00aa6444162b22fe2fb910b28bd5